lib/mail/message.ex

```diff
@@ -4,6 +4,14 @@ defmodule Mail.Message do
             parts: [],
             multipart: false
 
+  @type t ::
+          %__MODULE__{
```

Comment:
can you break these out into their own types? The types themselves can be similar to here, just that these types will be necessary for the functional specs. Similar to what I did here: https://github.com/DockYard/openid_connect/blob/master/lib/openid_connect.ex

Reply:
Thanks for the feedback. I broke out the struct fields into individual types.